test/e2e/provider/eip-6963.spec.js

```diff
@@ -0,0 +1,94 @@
+const { strict: assert } = require('assert');
+const FixtureBuilder = require('../fixture-builder');
+const { convertToHexValue, withFixtures, openDapp } = require('../helpers');
+
+// https://github.com/thenativeweb/uuidv4/blob/bdcf3a3138bef4fb7c51f389a170666f9012c478/lib/uuidv4.ts#L5
+const UUID_V4_REGEX =
+  /(?:^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[a-f0-9]{4}-[a-f0-9]{12}$)|(?:^0{8}-0{4}-0{4}-0{4}-0{12}$)/u;
+
+const SVG_DATA_URI_REGEX = /^data:image\/svg\+xml,/u;
+
+describe('EIP-6963 Provider', function () {
+  const ganacheOptions = {
+    accounts: [
+      {
+        secretKey:
+          '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+        balance: convertToHexValue(25000000000000000000),
+      },
+    ],
+  };
+
+  it('should respond to the request provider event', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await openDapp(driver);
+        await driver.executeScript(`
+          window.announceProviderEvents = []
+          window.addEventListener(
+            "eip6963:announceProvider",
+            (event) => {
+              window.announceProviderEvents.push(event)
+            }
+          );
+          window.dispatchEvent(new Event("eip6963:requestProvider"));
+        `);
+        const announceProviderEvents = await driver.executeScript(`
+          return window.announceProviderEvents.map(event => {
+            return {
+              type: event.type,
+              detail: {
+                ...event.detail,
+                provider: Boolean(event.detail.provider)
+              }
+            }
+          })
+        `);
+
+        assert.match(announceProviderEvents[0].detail.info.uuid, UUID_V4_REGEX);
+        delete announceProviderEvents[0].detail.info.uuid;
+        assert.match(
+          announceProviderEvents[0].detail.info.icon,
+          SVG_DATA_URI_REGEX,
+        );
+        delete announceProviderEvents[0].detail.info.icon;
+        assert.deepStrictEqual(announceProviderEvents, [
+          {
+            type: 'eip6963:announceProvider',
+            detail: {
+              info: {
+                name: 'MetaMask Main',
```

Comment:
shouldn't have a suffix here correct?

Reply:
Sorry, to clarify. For non-prod builds, the build type is included in the announced provider name, but suffixes like `MV3`, `lavamoat`, `snow` are not.

Reply:
prod, any build type: `MetaMask`
non-prod, build type MMI: `MetaMask Institutional`
non-prod, build type others: `MetaMask Main`, `MetaMask Beta`, `MetaMask Flask`, etc

@rekmarks is this what you had mind?

Reply:
I'm not sure what "Main" is, but AFAIK the possible values here should be:
- `MetaMask` (default build type, i.e. stable)
- `MetaMask Institutional` (for the MMI build type)
- `MetaMask Beta` (for the beta build type)
- `MetaMask Flask` (for the Flask build type)

Reply:
clarified via slack. This is fine as long as stable prod builds are `MetaMask`